tests/image/embedding/test_model.py

```diff
@@ -89,3 +89,46 @@ def test_vissl_training(backbone, training_strategy, head, pretraining_transform
 
     trainer = flash.Trainer(max_steps=3, max_epochs=1, gpus=torch.cuda.device_count())
     trainer.fit(embedder, datamodule=datamodule)
+
+
+@pytest.mark.skipif(not (_TORCHVISION_AVAILABLE and _VISSL_AVAILABLE), reason="vissl not installed.")
+@pytest.mark.parametrize(
+    "backbone, training_strategy, head, pretraining_transform",
+    [
+        ("vision_transformer", "simclr", "simclr_head", "simclr_transform"),
+        pytest.param(
+            "vision_transformer",
+            "dino",
+            "dino_head",
+            "dino_transform",
+            marks=pytest.mark.skipif(torch.cuda.device_count() < 1, reason="VISSL DINO calls all_reduce internally."),
+        ),
+        ("vision_transformer", "barlow_twins", "simclr_head", "barlow_twins_transform"),
+        ("vision_transformer", "swav", "swav_head", "swav_transform"),
+    ],
+)
+def test_vissl_prediction(backbone, training_strategy, head, pretraining_transform):
+    datamodule = ImageClassificationData.from_datasets(
+        train_dataset=FakeData(),
+        batch_size=4,
+    )
+
+    training_strategy_kwargs = {
+        "dims": [384, 2048, 2048, 256],
+    }
+    dim_key = "latent_embedding_dim" if training_strategy == "barlow_twins" else "embedding_dim"
+    training_strategy_kwargs[dim_key] = 256
+
+    embedder = ImageEmbedder(
+        backbone=backbone,
+        training_strategy=training_strategy,
+        head=head,
+        pretraining_transform=pretraining_transform,
+        training_strategy_kwargs=training_strategy_kwargs,
+    )
+
+    trainer = flash.Trainer(max_steps=3, max_epochs=1, gpus=torch.cuda.device_count())
+
+    predict_datamodule = ImageClassificationData.from_datasets(predict_dataset=FakeData(), batch_size=4)
+    trainer.fit(embedder, datamodule=datamodule)
+    trainer.predict(embedder, datamodule=predict_datamodule)
```

Comment:
Could have done this in the test that existed before but thought decoupling training test vs prediction test is a good idea?